src/combinators/chain.jl

```diff
@@ -0,0 +1,107 @@
+using ConcreteStructs
+using DynamicIterators
+using DynamicIterators: dub
+import Base
```

Comment:
What does this do?

Reply:
`ConcreteStructs` is pretty great:
https://github.com/jonniedie/ConcreteStructs.jl

Some of this is vestigial, I was using `dub` for a while before I realized an `Evolution` doesn't need it